presto-main/src/main/java/com/facebook/presto/operator/ExchangeClient.java

```diff
@@ -506,4 +513,27 @@ private static void closeQuietly(HttpPageBufferClient client)
             // ignored
         }
     }
+
+    private static class ExponentialMovingAverage
+    {
+        private final double alpha;
+        private double oldValue;
+
+        public ExponentialMovingAverage(double alpha, double initialValue)
+        {
+            this.alpha = alpha;
+            this.oldValue = initialValue;
+        }
+
+        public void addSample(double value)
```

Comment:
The interface is a little bit weird. The `addSample` and the constructor taskes `double`, but returns `long`. Should we use `long` everywhere?

Reply:
Changed types to long